src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -112,45 +112,57 @@ private struct CommandData
+                        try
+                        {
+                            await Database.TxMerger.Enqueue(mergedCmd);
+                        }
+                        catch (ConcurrencyException)
+                        {
+                            HttpContext.Response.StatusCode = (int) HttpStatusCode.Conflict;
+                            throw;
+                        }
 
-                    var waitForReplicasTimeout = GetTimeSpanQueryString("waitForReplicasTimeout", required: false);
-                    if (waitForReplicasTimeout != null)
-                    {
-                        await WaitForReplicationAsync(waitForReplicasTimeout.Value, mergedCmd);
-                    }
+                        var waitForReplicasTimeout = GetTimeSpanQueryString("waitForReplicasTimeout", required: false);
+                        if (waitForReplicasTimeout != null)
+                        {
+                            await WaitForReplicationAsync(waitForReplicasTimeout.Value, mergedCmd);
+                        }
 
-                    if (waitForIndexesTimeout != null)
-                    {
-                        await
-                            WaitForIndexesAsync(waitForIndexesTimeout.Value, mergedCmd.LastEtag,
-                                mergedCmd.ModifiedCollections);
-                    }
+                        if (waitForIndexesTimeout != null)
+                        {
+                            await
+                                WaitForIndexesAsync(waitForIndexesTimeout.Value, mergedCmd.LastEtag,
+                                    mergedCmd.ModifiedCollections);
+                        }
 
-                    HttpContext.Response.StatusCode = (int) HttpStatusCode.Created;
+                        HttpContext.Response.StatusCode = (int) HttpStatusCode.Created;
 
-                    using (var writer = new BlittableJsonTextWriter(readBatchCommandContext, ResponseBodyStream()))
-                    {
-                        readBatchCommandContext.Write(writer, new DynamicJsonValue
+                        using (var writer = new BlittableJsonTextWriter(readBatchCommandContext, ResponseBodyStream()))
                         {
-                            ["Results"] = mergedCmd.Reply
-                        });
+                            readBatchCommandContext.Write(writer, new DynamicJsonValue
+                            {
+                                ["Results"] = mergedCmd.Reply
+                            });
+                        }
+                    }
+                    finally
+                    {
+                        //since the usage of this object does not end with TxMerger,
+                        //it is wrong to have it implement IDisposable.
+                        //Still, we need to dispose the BlittableJsonReaderObject objects that are in this command
+                        mergedCmd?.ManualDispose();
```

Comment:
We won't do that, this is really ugly an not necessary.
We can ask the merger not to dispose this one.